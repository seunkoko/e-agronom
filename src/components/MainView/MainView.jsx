// react library
import React from 'react';

// styles
import './MainView.scss';

/**
 * Renders MainView component
 * @param {object} props - component props
 * @returns {JSX} JSX
 */
const MainView = ({
  pending,
  categoryImages,
  handleLoadMore,
  loadMorePending,
}) => (
  <div id="main-view">
    {
      pending && <div className="loader"></div>
    }

    {
      !pending && categoryImages && categoryImages.map((categoryImage, index) => {
        const data = <div
          key={index}
          className="cat-display"
        >
          <img
            className="image-display"
            src={categoryImage.url}
            alt="img"
          />
        </div>
        return data;
      })
    }

    {
      !pending && categoryImages && categoryImages.length <= 0 && (
        <div>Sorry!!! No cats found</div>
      )
    }

    {
      !pending && categoryImages && categoryImages.length > 0 && (
        <div>
          <button
            type="button"
            className="load-more-btn"
            onClick={handleLoadMore}
          >
            {
              !loadMorePending && <p>Load more Images...</p>
            }

            {
              loadMorePending && <div className="loader loader-pending"></div>
            }
          </button>
        </div>
      )
    }
  </div>
);

export default MainView;
