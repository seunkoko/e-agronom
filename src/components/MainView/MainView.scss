@import "../../stylesheets/variables";

#main-view {
  width: inherit;
  margin-left: 20%;
  display: grid;
  grid-template-columns: 49% 49%;
  grid-column-gap: 2%;
  grid-row-gap: 3%;
  padding: 30px 20px;

  @media (max-width: $laptop-width) {
    margin-left: 25%;
  }

  @media (max-width: $mobile-width) {
    margin-left: 32%;
    display: flex;
    flex-direction: column;
  }

  @media (max-width: $phone-width) {
    margin-left: 20%;
    display: flex;
    flex-direction: column;
  }
}

.cat-display {
  height: 450px;
  cursor: pointer;

  @media (max-width: $mobile-width) {
    margin-bottom: 50px;
  }

  @media (max-width: $phone-width) {
    height: 250px;
  }
}

.image-display {
  width: 90%;
  border-radius: 8%;
  padding: 3%;
  height: 100%;

  &:hover {
    border: 1.5px dotted $primary-orange;
  }
}
