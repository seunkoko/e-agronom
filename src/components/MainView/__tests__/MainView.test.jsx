// react library
import React from 'react';

// third-party libraries
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

// components
import MainView from '../MainView';

describe('MainView component', () => {
  let wrapper;
  let props = {
    pending: false,
    categoryImages: [],
    handleLoadMore: jest.fn(),
    loadMorePending: false,
  }

  beforeEach(() => {
    wrapper = shallow(
      <MainView { ...props } />
    );
  });

  it('renders correctly', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when pending is true', () => {
    props = {
      ...props,
      pending: true,
    }

    wrapper = shallow(
      <MainView { ...props } />
    );
    expect(wrapper).toBeTruthy();
  });

  it('renders correctly when category images are provided', () => {
    props = {
      ...props,
      pending: false,
      categoryImages: [
        {
          url: 'fake-url1'
        },
        {
          url: 'fake-url2'
        }
      ],
    }

    wrapper = shallow(
      <MainView { ...props } />
    );
    expect(wrapper).toBeTruthy();
  });

  it('renders correctly when loadMorePending is true', () => {
    props = {
      ...props,
      loadMorePending: true,
    }

    wrapper = shallow(
      <MainView { ...props } />
    );
    expect(wrapper).toBeTruthy();
  });
});
