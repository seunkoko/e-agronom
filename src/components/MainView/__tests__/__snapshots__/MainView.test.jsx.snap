// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainView component renders correctly 1`] = `
<div
  id="main-view"
>
  <div>
    Sorry!!! No cats found
  </div>
</div>
`;
